full-stack-practice/client/lib/read.ts

```diff
@@ -0,0 +1,9 @@
+import { products, type Product } from './data';
+
+export function readCatalog(): Promise<Product[]> {
+  return Promise.resolve(products);
```

Comment:
This is not calling the backend server, so this is not a full-stack application. This must be rewritten to use `fetch` to call the server. Same for the `readProduct` function below.